src/OrleansRuntime/Streams/PersistentStream/PersistentStreamPullingAgent.cs

```diff
@@ -555,17 +563,8 @@ private void StartInactiveCursors(StreamConsumerCollection streamData)
                 }
             }
             StreamHandshakeToken newToken = await batchDeliveryTask;
-            if (newToken != null)
-            {
-                consumerData.LastToken = newToken;
-                consumerData.Cursor = queueCache.GetCacheCursor(consumerData.StreamId.Guid,
-                    consumerData.StreamId.Namespace, newToken.Token);
-            }
-            else
-            {
-                consumerData.LastToken = StreamHandshakeToken.CreateDeliveyToken(batch.SequenceToken); // this is the currently delivered token
-            }
-
+            consumerData.LastToken = StreamHandshakeToken.CreateDeliveyToken(batch.SequenceToken); // this is the currently delivered token
```

Comment:
Better to move that line also to outside that function, just like you did to the rest of the code.